src/modules/producer.js

```diff
@@ -50,28 +69,31 @@ function createRpcQueue(queue) {
  */
 function maybeAnswer(queue) {
   var rpcQueue = amqpRPCQueues[queue];
-
+  
   return (msg) => {
     //check the correlation ID sent by the initial message using RPC
     var corrId = msg.properties.correlationId;
-
+    
     try {
       //if we found one, we execute the callback and delete it because it will never be received again anyway
       rpcQueue[corrId].resolve(parsers.in(msg));
       this.conn.config.transport.info('bmq:producer', '[' + queue + '] < answer');
       delete rpcQueue[corrId];
-    } catch(e) {
+    } catch (e) {
       this.conn.config.transport.error(new Error('Receiving RPC message from previous session: callback no more in memory. ' + queue));
     }
   };
 }
 
 function publishOrSendToQueue(queue, msg, options) {
-  if (!options.routingKey) {
-    return this.channel.sendToQueue(queue, msg, options);
-  } else {
-    return this.channel.publish(queue, options.routingKey, msg, options);
-  }
+  return checkQueue(this.channel, queue)
```

Comment:
@mrister how is that possible? It means that amqp.node caches the existance of the queue?
Do your queues exists on your sample code?